### DISCUSSION TITLE
Unable to download artifacts via Github API

### DISCUSSION THREAD

**Discussion by mohanarpit** [community/community#24305] [Apps, API and Webhooks]
<p>I’m using the Github API to download artifacts for a particular workflow’s run. I get the error:<br>
<strong>"Failed to generate URL to download artifact"</strong></p>
<p>Although, when I list the artifacts for the workflow, I can see the artifact listed in the API response.</p>
<pre><code class="lang-auto">{
    "id": 43821968,
    "node_id": "MDg6QXJ0aWZhY3Q0MzgyMTk2OA==",
    "name": "workflow-package",
    "size_in_bytes": 13559065,
    "url": "https://api.github.com/repos/mohanarpit/appsmith/actions/artifacts/43821968",
    "archive_download_url": "https://api.github.com/repos/mohanarpit/appsmith/actions/artifacts/43821968/zip",
    "expired": true,
    "created_at": "2021-02-28T10:55:10Z",
    "updated_at": "2021-02-28T10:55:10Z",
    "expires_at": "2021-03-01T10:54:19Z"
}
</code></pre>
<p>Can someone please help me debug this?</p>

**Comment by torbjornvatn** [comment]
<p>Hi <a class="mention" href="/mohanarpit">@mohanarpit</a>, I’m pretty sure it’s related to this <a href="https://github.community/t/bug-github-api-list-workflow-runs-returns-0-runs-even-though-there-are-runs/165080" class="inline-onebox">[Bug] GitHub API - List workflow runs returns 0 runs even though there are runs</a></p>
<p>Seems like a bug on Github’s end</p>

**Comment by mohanarpit** [comment]
<p>Thanks for your response. But I think this is a different bug. I see the workflow runs being listed. It’s <strong>not</strong> an empty array for me. It’s just that none of the artifact links listed in the runs work.</p>

**Comment by mohanarpit** [comment, marked as answer]
<p>The bug seems to be fixed &amp; workflow artifacts seem to be working for new runs now. Thanks!</p>
